### PR TITLE
Milestone 2 - Joseph Hammana

### PR DESCRIPTION
**Pull request by Hammana963**
# Milestone: 2
# Developer: Joseph Hammana

### Pull Request Summary

This is a new pull request for turning in Milestone 2

### Pull Request Checklist
- [x] Code is neat, readable, and works
- [x] Comments are appropriate
- [x] The commit message follows our guidelines
- [x] The milestone number is specified
- [x] The developer name is specified
- [x] The summary is completed

### Reviewer Checklist  - PULL REQUEST REVIEWER ONLY

IMPORTANT: The rest of the sections in this checklist should only be filled out by authorized pull request reviewers. If you are the individual template contributor, do not fill out the rest of the fields or check the boxes.

NOTE: Milestones can only be completed when all boxes are checked.

- [x] The code is fully reviewed
- [x] Meaningful feedback is given
- [x] Comment that you have reviewed the code
- [x] This box is checked

### CODE REVIEW

File: milestone2.ts [hunk 1]
```diff
@@ -83,32 +83,64 @@ const userData: Volunteer[] = [
...
 */
-let findIndexAns: GetNumber; // Code here
+let findIndexAns: GetNumber = (data) => {  // Code here
+  for(let i=0; i<data.length; i++){
+    if (data[i].city === "SF") {
+      return i;
+    }
+  }
+
+  return -1;
```
Comment (sfwathen) [inline milestone2.ts]: Can you redo this question using data handling functions like specified in the question. If you need a reference on data handling functions, look at slide 9 of [Bootcamp session 4](https://docs.google.com/presentation/d/1Jdvx7Ji4n3m6jODDX2N_gL5BO-ZTEguSNa_drSd5s1g/edit?usp=sharing)

File: milestone2.ts [hunk 2]
```diff
@@ -83,32 +83,64 @@ const userData: Volunteer[] = [
...
-let findCAOverN: GetVolunteers; // Code here
+let findCAOverN: GetVolunteers = (data, minNumber) => { // Code here
+  let resData: Volunteer[] = []
+  for(let i=0; i<data.length; i++){
+    if (data[i].state === "CA" && data[i].age > minNumber) {
+      resData.push(data[i]);
+    }
+  }
+  return resData;
+};
```
Comment (sfwathen) [inline milestone2.ts]: redo with data handling functions

File: milestone2.ts [hunk 3]
```diff
@@ -83,32 +83,64 @@ const userData: Volunteer[] = [
...
-let findSBStaff: GetVolunteer; // Code here
+let findSBStaff: GetVolunteer = (data) => {
+  let resPerson: Volunteer = data[0];
+  for(let i=0; i<data.length; i++){
+    if (data[i].city === "SB") {
+      return (data[i]);
+    }
+  }
+  return resPerson;
+}; // Code here
```
Comment (sfwathen) [inline milestone2.ts]: redo with data handling functions

File: milestone2.ts [hunk 4]
```diff
@@ -163,9 +199,18 @@ let getVolunteerInfo: GetVolunteerInfo = (vol) => {
...
+let daBigTest: GetVolunteer = (data) => {
+  let resPerson: Volunteer = data[0];
+  for(let i=0; i<data.length; i++){
+    if (data[i].city === "SLO" && data[i].age > 40) {
+      data[i].position = "staff";
+      return data[i];
+
+    }
+  }
+}; // Code here
```
Comment (sfwathen) [inline milestone2.ts]: redo with data handling functions (make sure to handle the case where you don't find a volunteer with those criteria)

**Review by sfwathen — CHANGES_REQUESTED** [review]
Hi Joseph, 
Can you resubmit this milestone using data handling functions as asked for in the questions. For reference, we talked about data handling functions in [Bootcamp session 4](https://h4i.notion.site/Typescript-NPM-b02ee93561844b889a072541d65044d4), specifically on slide 9 of the slide deck. 
Thanks!

File: milestone2.ts (hunk 4 above)
Comment (Hammana963) [reply milestone2.ts]: I have just resubmitted with the data handling function corrections made.

**Review by sfwathen — COMMENTED** [review]
Hey Joseph, thanks for completing the questions with data handling functions. Everything looks good now so I'll check you off for this milestone!

**Pull request closed without merging**